src/main/java/emu/grasscutter/game/inventory/Inventory.java

```diff
@@ -138,26 +138,23 @@ public void addItems(Collection<GameItem> items) {
         this.addItems(items, null);
     }
 
-    public void addItems(Collection<GameItem> items, ActionReason reason) {
-        List<GameItem> changedItems = new LinkedList<>();
 
-        for (GameItem item : items) {
+    public void addItems(Collection<GameItem> items, ActionReason reason) {
+        List<GameItem> changedItems = new ArrayList<>();
+        for (var item : items) {
+            if (item.getItemId() == 0) continue;
```

Comment:
`putItem` shouldn't have any issues rejecting invalid items so this is probably unnecessary .